[Analyzer] Ensure plugin sources are added to the sources manager

amorde opened the pull request:
Closes #8540

@fabiopelosin could you try this out on your project to confirm that it resolves the issue you are experiencing?

dnkoutso commented on CHANGELOG.md:
```diff
@@ -38,7 +38,9 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
 ##### Bug Fixes
 
-* None.  
+* Fix a bug that prevented dependencies in a plugin source from resolving  
```
this not on the right spot.

amorde replied:
ah my bad, good catch

segiddins commented on lib/cocoapods/installer/analyzer.rb:
```diff
@@ -170,6 +170,9 @@ def sources
           unless plugin_sources.empty?
             result.insert(0, *plugin_sources)
+            plugin_sources.each do |source|
+              config.sources_manager.add_source(source)
```
should this be added to the global sources manager? should we make our own sources manager in the analyzer?

amorde replied:
updated this to use a separate sources manager rather than the global one

amorde commented:
Rebased, pointed to master (this was originally intended to be released as a minor 1.6.x release),  and updated the analyzer and resolver to handle separate source managers

amorde commented:
Yay green again

amorde commented:
I was (and still am) very confused, not sure I am deserving of the blind trust 😂 until someone comes up with an easy to reproduce bug _or_ the original bug author tests out the change, I guess we'll go with the test cases here
